src/openzaak/components/besluiten/tests/admin/test_besluit_audittrail.py

```diff
@@ -0,0 +1,159 @@
+            "toelichting": "desc",
+        }
+        data.update(inline_data)
+
+        self.client.post(add_url, data)
+
+        self.assertEqual(Besluit.objects.count(), 1)
+
+        besluit = Besluit.objects.get()
+        besluit_url = get_operation_url("besluit_read", uuid=besluit.uuid)
+
+        self.assertEqual(AuditTrail.objects.count(), 1)
+
+        audittrail = AuditTrail.objects.get()
+
+        self.assertEqual(audittrail.bron, "BRC")
+        self.assertEqual(audittrail.actie, "create")
+        self.assertEqual(audittrail.resultaat, 0)
+        self.assertEqual(audittrail.applicatie_weergave, "admin")
+        self.assertEqual(audittrail.gebruikers_id, f"{self.user.id}"),
+        self.assertEqual(audittrail.gebruikers_weergave, self.user.get_full_name()),
+        self.assertEqual(audittrail.hoofd_object, f"http://testserver{besluit_url}"),
+        self.assertEqual(audittrail.resource, "besluit"),
+        self.assertEqual(audittrail.resource_url, f"http://testserver{besluit_url}"),
+        self.assertEqual(audittrail.resource_weergave, besluit.unique_representation()),
+        self.assertEqual(audittrail.oud, None)
+
+        new_data = audittrail.nieuw
+        self.assertEqual(new_data["toelichting"], "desc")
+
+    def test_change_besluit(self):
+        besluit = BesluitFactory.create(toelichting="old")
+        besluit_url = get_operation_url("besluit_read", uuid=besluit.uuid)
+        change_url = reverse("admin:besluiten_besluit_change", args=(besluit.pk,))
+        data = {
+            "uuid": besluit.uuid,
+            "_besluittype": besluit._besluittype.id,
+            "verantwoordelijke_organisatie": besluit.verantwoordelijke_organisatie,
+            "datum": besluit.datum,
+            "ingangsdatum": "15-11-2019",
+            "toelichting": "new",
+        }
+        data.update(inline_data)
+
+        self.client.post(change_url, data)
+
+        self.assertEqual(AuditTrail.objects.count(), 1)
+
+        besluit.refresh_from_db()
+        audittrail = AuditTrail.objects.get()
+
+        self.assertEqual(audittrail.bron, "BRC")
+        self.assertEqual(audittrail.actie, "update")
+        self.assertEqual(audittrail.resultaat, 0)
+        self.assertEqual(audittrail.applicatie_weergave, "admin")
+        self.assertEqual(audittrail.gebruikers_id, f"{self.user.id}"),
+        self.assertEqual(audittrail.gebruikers_weergave, self.user.get_full_name()),
+        self.assertEqual(audittrail.hoofd_object, f"http://testserver{besluit_url}"),
+        self.assertEqual(audittrail.resource, "besluit"),
+        self.assertEqual(audittrail.resource_url, f"http://testserver{besluit_url}"),
+        self.assertEqual(audittrail.resource_weergave, besluit.unique_representation()),
+
+        old_data, new_data = audittrail.oud, audittrail.nieuw
+        self.assertEqual(old_data["toelichting"], "old")
+        self.assertEqual(new_data["toelichting"], "new")
+
+    def test_delete_besluit_action(self):
```

Comment:
This means that the AuditTrail remains in the database, even after the main resource (`Zaak` in this case) is deleted right? Where would you be able to see the audittrail for this action, if the resource is already deleted?

@joeribekker @sergei-maertens currently in the audittrail in vng-api-common the audittrail is deleted if the main resource is deleted via the API, is this okay?

Reply:
> This means that the AuditTrail remains in the database, even after the main resource (Zaak in this case) is deleted right? Where would you be able to see the audittrail for this action, if the resource is already deleted?

This is indeed not correct. The audit trail should be deleted if the main resource is deleted.

> @joeribekker @sergei-maertens currently in the audittrail in vng-api-common the audittrail is deleted if the main resource is deleted via the API, is this okay?

Yes. Delete really means destroy (even if I don't like this from a developer perspective). It's the law :)